src/qibo/derivative.py

```diff
@@ -9,21 +9,35 @@ def parameter_shift(
     hamiltonian,
     parameter_index,
     initial_state=None,
+    scale_factor=None,
 ):
     """In this method the parameter shift rule (PSR) is implemented.
     Given a circuit U and an observable H, the PSR allows to calculate the derivative
     of the expected value of H on the final state with respect to a variational
     parameter of the circuit.
+    There is also the possibility of setting a scale factor. It is useful when a
+    circuit's parameter is obtained by combination of a variational
+    parameter and an external object, such as a training variable in a Quantum
+    Machine Learning problem. For example, performing a re-uploading strategy
+    to embed some data into a circuit, we apply to the quantum state rotations
+    whose angles are in the form: theta' = theta * x, where theta is a variational
+    parameter and x an input variable. The PSR allows to calculate the derivative
+    with respect of theta' but, if we want to optimize a system with respect its
+    variational parameters we need to "free" this procedure from the x depencency.
+    If the `scale_factor` is not provided, it is set equal to one and doesn't
+    affect the calculation.
 
     Original references:
         `https://arxiv.org/abs/1811.11184`;
         `https://arxiv.org/abs/1803.00745`.
```

Comment:
More relevant for the docs added in #741 but if you like there is a way in Sphinx to make these appear as actual links in the docs page (so that one can press and go directly to the arxiv page). Now one needs to copy-paste.